File: backend/src/main/java/com/igreja/site/model/Evento.java
package com.igreja.site.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;

@Entity
@Table(name = "eventos")
public class Evento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotBlank
    private String titulo;

    @NotNull
    private LocalDate data;

    private String horario;

    private String local;

    @Column(length = 3000)
    private String descricao;

    private String imagemUrl;

    // true = evento ja realizado (aparece em "Eventos que a igreja realizou")
    // false = evento futuro (aparece em "Proximos eventos")
    private boolean realizado;

    public Evento() {}

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getTitulo() { return titulo; }
    public void setTitulo(String titulo) { this.titulo = titulo; }

    public LocalDate getData() { return data; }
    public void setData(LocalDate data) { this.data = data; }

    public String getHorario() { return horario; }
    public void setHorario(String horario) { this.horario = horario; }

    public String getLocal() { return local; }
    public void setLocal(String local) { this.local = local; }

    public String getDescricao() { return descricao; }
    public void setDescricao(String descricao) { this.descricao = descricao; }

    public String getImagemUrl() { return imagemUrl; }
    public void setImagemUrl(String imagemUrl) { this.imagemUrl = imagemUrl; }

    public boolean isRealizado() { return realizado; }
    public void setRealizado(boolean realizado) { this.realizado = realizado; }
}

File: backend/src/models/Evento.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/database');

// Equivalente a Evento.java
const Evento = sequelize.define(
  'Evento',
  {
    id: { type: DataTypes.INTEGER, primaryKey: true, autoIncrement: true },
    titulo: { type: DataTypes.STRING, allowNull: false, validate: { notEmpty: true } },
    data: { type: DataTypes.DATEONLY, allowNull: false },
    horario: { type: DataTypes.STRING, allowNull: true },
    local: { type: DataTypes.STRING, allowNull: true },
    descricao: { type: DataTypes.STRING(3000), allowNull: true },
    imagemUrl: { type: DataTypes.STRING, allowNull: true },
    // true = evento ja realizado (aparece em "Eventos que a igreja realizou")
    // false = evento futuro (aparece em "Proximos eventos")
    realizado: { type: DataTypes.BOOLEAN, allowNull: false, defaultValue: false },
  },
  {
    tableName: 'eventos',
    timestamps: false,
  }
);

module.exports = Evento;
